src/main/java/euteam/cleanmachine/model/facility/MachineState.java

```diff
@@ -0,0 +1,20 @@
+package euteam.cleanmachine.model.facility;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public abstract class MachineState {
```

Comment:
Looks like this has to be an Interface: https://github.com/iluwatar/java-design-patterns/tree/master/state

Reply:
Can we change it to an interface? Last time we tried to put an interface into the db with jpa it didn't work.

Reply:
Oh yeah you're right, I think we have to keep it as a class to save it in the database... My bad